server/controllers/UserController.js

```diff
@@ -0,0 +1,38 @@
+import models from '../database/models';
+import helpers from '../helpers';
+
+const { Book, LendingHistory, Author } = models;
+const { responseMessage } = helpers;
+
+/**
+ * @class UserController
+ */
+export default class UserController {
+  /**
+   * @description allows users to borrow a book
+   * @param {object} request express request object
+   * @param {object} response express response object
+   * @returns {json} json
+   * @memberof UserController
+   */
+  static async borrowBook(request, response) {
+    try {
+      const { id } = request.userData;
+      const { duration, bookId } = request.body;
+      const existingBook = await Book.findOne({ where: { id: bookId }, include: [Author] });
+      if (!existingBook) return responseMessage(response, 404, { message: 'book not found' });
+      const { dataValues } = existingBook;
+      const bookStock = dataValues.stock;
+      const { fullname } = dataValues.Author;
+      delete dataValues.authorId;
+      delete dataValues.Author;
+      if (!bookStock) return responseMessage(response, 200, { status: 'success', message: 'book is out stock', data: { ...dataValues, authorName: fullname } });
+      await LendingHistory.create({ duration, bookId, userId: id });
+      await Book.update({ stock: bookStock - 1 }, { where: { id: bookId } });
+      return responseMessage(response, 201, { status: 'success', message: 'book successfully borrowed', data: { ...dataValues, authorName: fullname } });
```

Comment:
what about if the stock is one and you subtract one and it turns to zero?

Reply:
When it turns to zero, it has been taken care of above, it would give a response book is out of stock

Reply:
when creating the lending history you didn't add a default status in the lending history table